lib/fastly_nsq/rake_task.rb

```diff
@@ -44,7 +44,11 @@ def run_tasks
         end
       end
 
-      listener_threads.map(&:join)
+      while listener_threads.any?(&:status) do
+        listener_threads.each do |thread|
+          thread.join(1)
+        end
+      end
```

Comment:
Is checking status here necessary? As long as any one thread has a truthy `status` it will still try to join all the threads we are tracking, even the threads that have exited. I guess it is so that you can try to join every second until all threads are joined. Super tiny race condition where all threads exit before any it reaches this block but that seems unlikely/impossible.